### PR TITLE
Add .git-blame-ignore-revs

### PR DESCRIPTION
**Pull request by katcharov**
See:

* https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view
* https://github.com/orgs/community/discussions/5033

This ignores some reformatting commits arising from DRIVERS-2789 when viewing git blame on GitHub and in other tools. Tested `auth.md` in IntelliJ.

Blame [before](https://github.com/mongodb/specifications/blame/b5a36a7d5c91a3ac708450ec50d05c0ed6bd846a/source/auth/auth.md#L10), note most lines attributed to the `Sync with 4b7d757` commit.

Blame [after](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5/source/auth/auth.md#L10) (as of this change). In [some cases](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5/source/auth/auth.md#L196), tools do not identify the correct commit.

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+
+# DRIVERS-2789
+ca372f0c6a1353742437d65131b37544492692c2
```
Comment (eramongodb) [inline .git-blame-ignore-revs:3]: Recommend using `git log -n 1 --format=format:"%H # %s (%cs)" <hash>` to generate entries for this file. e.g.:

```
ca372f0c6a1353742437d65131b37544492692c2 # Sync with 4b7d7573 (2024-02-26)
634d6399bc166d27f290599d8facf034fa17be91 # Sync with 4b7d7573 (2024-02-26)
405e736abf6884c69669fcc060ef76c44f2e63e7 # Sync with 4b7d7573 (2024-02-26)
...
cea087babea68571fa4068ace6b2d326c3a0a22d # DRIVERS-2789 Convert BSON Corpus Spec to Markdown (#1499) (2024-01-23)
e9ae09da918dc7ca7e5ae12a1af6ca34e9c7854e # DRIVERS-2790 Add codespell checker (#1491) (2024-01-10)
eef59d9d386bb054e0eb59e6bd540a030186f899 # DRIVERS-2789 Use Markdown for Specifications Documentation (#1482) (2024-01-04)
```

Comment (katcharov) [reply .git-blame-ignore-revs]: Used `cat .git-blame-ignore-revs | xargs -n1 git log -n 1 --format=format:"%H # %s (%cs)" $1` to update.

**Comment by eramongodb** [comment]
I'd like to defer my approval/disapproval until more feedback is given by other reviewers.

To reiterate my comments from external discussion in short summary, I would rather deal with the presence of the Markdown conversion and the "Sync with" commits in the line-based git blame history over allowing for the possibility of misleading git blame line-to-commit attribution resulting from the use of `--ignore-rev` on commits with significant change in content. The conversion from reStructuredText to Markdown alone is significant enough for me to be reluctant to ignoring them. The inclusion of "Sync with" commits that restore git blame history, which may _also_ contain important spec changes (e.g. https://github.com/mongodb/specifications/commit/a2886304afb006cc39808cd67a3503cc1b221116) unrelated to Markdown conversion, further strengthens my reluctance.

Examples of the concerning line-based git blame behavior with ignored refs include (note: use of `~` suffix for hash in URL to disable ignoring revs):

* [Line 196 of auth.md](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5/source/auth/auth.md#L196). Incorrectly attributed to unrelated commit [5112bcc](https://github.com/mongodb/specifications/commit/5112bcca7789bb274c91dc425078508751fee142) (April 2023) when it should be:
  * [ad2d7a5](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5~/source/auth/auth.md#L196) (blame without ignore revs): blame prior to change -> [eb2e511](https://github.com/mongodb/specifications/blame/eb2e511137acc5d6745a29dafec3222ec08971f1/source/auth/auth.md#L240)
  * [eb2e511](https://github.com/mongodb/specifications/blame/eb2e511137acc5d6745a29dafec3222ec08971f1/source/auth/auth.md#L240): select annotated revision -> [c7494b9](https://github.com/mongodb/specifications/commit/c7494b999268c1f583140929fd4a4b55e01f88da#diff-f123e62b6be2679702be737483de4686c660a64ce405ba66a9a240b89945f138R177)
  * [c7494b9](https://github.com/mongodb/specifications/commit/c7494b999268c1f583140929fd4a4b55e01f88da#diff-f123e62b6be2679702be737483de4686c660a64ce405ba66a9a240b89945f138R177): the correct commit (March 2018).
* [Line 395 of change-streams.md](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5/source/change-streams/change-streams.md#L395). Incorrectly attributed to [35e466d](https://github.com/mongodb/specifications/commit/35e466ddf25059cb30e4113de71cdebd3754657f) (May 2018) when it should be:
  * [ad2d7a5](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5~/source/change-streams/change-streams.md#L395) (blame without ignore revs): blame prior to change -> [dc9394b](https://github.com/mongodb/specifications/blame/dc9394b1942eebcdf39c97023c57cc23fe4e7600/source/change-streams/change-streams.md#L432)
  * [dc9394b](https://github.com/mongodb/specifications/blame/dc9394b1942eebcdf39c97023c57cc23fe4e7600/source/change-streams/change-streams.md#L432): select annotated revision -> [8978899](https://github.com/mongodb/specifications/commit/89788990924ac3bfe43586233dd4f0159e2bf9b9#diff-b1cdd6bd41c94cc2b4ba9bef6f89e372fcc59b67dc96f349a95a85f9178095d4R364)
  * [8978899](https://github.com/mongodb/specifications/commit/89788990924ac3bfe43586233dd4f0159e2bf9b9#diff-b1cdd6bd41c94cc2b4ba9bef6f89e372fcc59b67dc96f349a95a85f9178095d4R364): the correct commit (April 2022).
* [Line 364 of unified-test-format.md](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5/source/unified-test-format/unified-test-format.md#L364). Incorrectly attributed to [ed4e62b](https://github.com/mongodb/specifications/commit/ed4e62b48939378b814cbd7d9d4939a2c1076a6b) (April 2021) when it should be:
  * [ad2d7a5](https://github.com/mongodb/specifications/blame/ad2d7a54c6dc5e5c52d42478f75f97e61c7c27f5~/source/unified-test-format/unified-test-format.md#L364) (blame without ignore revs): "blame prior to change" -> [a2efcb9](https://github.com/mongodb/specifications/blame/a2efcb91208dfef1626b34cdaffa6d3e1c844313/source/unified-test-format/unified-test-format.md)
  * [a2efcb9](https://github.com/mongodb/specifications/blame/a2efcb91208dfef1626b34cdaffa6d3e1c844313/source/unified-test-format/unified-test-format.md): line is missing because [it came from the "Sync with" commit](https://github.com/mongodb/specifications/commit/634d6399bc166d27f290599d8facf034fa17be91#diff-0f874d282835bde8ccd8e9f7920996c641233f6885b2a4a41cb1bb9ba644756fR364) -> blame the file from the commit referenced by the "Sync with" commit instead -> [4b7d757](https://github.com/mongodb/specifications/blame/4b7d7573e2635f68039ae298a462a1844f44eead/source/unified-test-format/unified-test-format.md#L364) (dead-end due to broken git blame history)
  * [4b7d757](https://github.com/mongodb/specifications/blame/4b7d7573e2635f68039ae298a462a1844f44eead/source/unified-test-format/unified-test-format.md#L364): select annotated revision (which broke git blame history) -> [f017041](https://github.com/mongodb/specifications/commit/f017041be26efa7a73edff0ffd47b158bd276370) -> blame the `.rst` file from the parent commit (prior to deletion) -> [bcde8cd](https://github.com/mongodb/specifications/blame/bcde8cd9052fb667f5530cfb4df9722f7f531c4a/source/unified-test-format/unified-test-format.rst#L462)
  * [bcde8cd](https://github.com/mongodb/specifications/blame/bcde8cd9052fb667f5530cfb4df9722f7f531c4a/source/unified-test-format/unified-test-format.rst#L462): select annotated revision -> [a288630](https://github.com/mongodb/specifications/commit/a2886304afb006cc39808cd67a3503cc1b221116#diff-f718c246f51e05b3731a7b03b176a0d4b0c381a9261810742d2677823f963f98R462)
  * [a288630](https://github.com/mongodb/specifications/commit/a2886304afb006cc39808cd67a3503cc1b221116#diff-f718c246f51e05b3731a7b03b176a0d4b0c381a9261810742d2677823f963f98R462): the correct commit (Jan 2024)

I recommend other reviewers examine the quality of line-based git blame history of affected files when considering the proposed changes in this PR.

**Comment by blink1073** [comment]
I agree with @eramongodb's assessment.

**Pull request closed without merging**